### PR TITLE
Fix BDC remember password and reprompting connection

### LINKED ISSUE #7921: not able to connect to saved big data cluster
OS: MacOS

Steps:
1. connect to a big data cluster and choose not to save the password
1. reload window
1. right click on the cluster just got added and choose Manage.

Expected:
The connect dialog should show up asking for password

Actual:
dialog not showing up and no logs in the console

### PR DESCRIPTION
**Pull request by Charles-Gagnon**
Fixes #7921

Fixed a couple of issues.

First off was the issue that even if you left remember password unchecked the node would still "remember" it and thus not prompt you for the password when you opened the dashboard. But if you restarted ADS then it would correctly not store it and so would fail to open the dashboard.

I've changed it to not save the password to the credential store, and then to re-prompt for the password if it doesn't exist (or the stored password is wrong, such as if it changed). The password will persist for the session though so the user won't be prompted unless an error occurs. 

Other fixes :

- Fixed error message displayed on Connection dialog to be more friendly (before it mentioned something about endpoints which isn't helpful in that scenario)
- Made dashboard open when new controller connection is created
- Added error handling for model
- Added reprompt connection for model if there's no valid controller (for example if the user cancelled out of the connection dialog when the dashboard was opened)

Follow-up items not included in this PR :
- Still need to make the dashboard indicate an error occurred (right now it just sits there spinning)

### CODE REVIEW

**Comment by Charles-Gagnon** [comment]
@kevcunnane @alanrenmsft Could you take a look at this when you get a chance?

File: extensions/big-data-cluster/src/bigDataCluster/dialog/hdfsDialogBase.ts [hunk 1]
```diff
@@ -28,6 +28,12 @@ export interface HdfsDialogProperties {
 	password?: string;
 }
 
+export class HdfsDialogCancelledError extends Error {
+	constructor(message: string = 'Dialog cancelled') {
```
Comment (kevcunnane) [inline extensions/big-data-cluster/src/bigDataCluster/dialog/hdfsDialogBase.ts]: Does this need localization?

Comment (Charles-Gagnon) [reply extensions/big-data-cluster/src/bigDataCluster/dialog/hdfsDialogBase.ts]: Depends. I generally go the route of "error messages shouldn't be exposed to the user and so don't need to be localized". But I also know we don't do that everywhere so if we think it's appropriate to default to localized errors I can switch it. I just don't want to make debugging a pain by having localized error messages in the console logs.

File: extensions/big-data-cluster/src/bigDataCluster/dialog/bdcDashboardModel.ts [hunk 2]
```diff
@@ -21,12 +22,22 @@ export class BdcDashboardModel {
...
 	public onDidUpdateBdcStatus = this._onDidUpdateBdcStatus.event;
+	public onError = this._onError.event;
 
 	constructor(private options: BdcDashboardOptions, ignoreSslVerification = true) {
-		this._clusterController = new ClusterController(options.url, options.auth, options.username, options.password, ignoreSslVerification);
-		this.refresh();
+		try {
+			this._clusterController = new ClusterController(options.url, options.auth, options.username, options.password, ignoreSslVerification);
+			this.refresh();
+		} catch {
```
Comment (alanrenmsft) [inline extensions/big-data-cluster/src/bigDataCluster/dialog/bdcDashboardModel.ts]: what error are you trying to catch here, the new ClusterController()? this.refresh is async and the error won't be caught here

File: extensions/big-data-cluster/src/bigDataCluster/dialog/bdcDashboardModel.ts [hunk 3]
```diff
@@ -46,21 +57,28 @@ export class BdcDashboardModel {
...
-				this._endpointsLastUpdated = new Date();
-				this._onDidUpdateEndpoints.fire(this.serviceEndpoints);
-			})
-		]).catch(error => {
-			showErrorMessage(error);
-		});
+		try {
+			if (!this._clusterController) {
+				// If this succeeds without error we know we have a clusterController at this point
+				await this.promptReconnect();
```
Comment (alanrenmsft) [inline extensions/big-data-cluster/src/bigDataCluster/dialog/bdcDashboardModel.ts]: what happens if user closes the connect dialog, it seems we are handling that scenario here.

File: extensions/big-data-cluster/src/bigDataCluster/dialog/bdcDashboardModel.ts (hunk 2 above)
Comment (Charles-Gagnon) [reply extensions/big-data-cluster/src/bigDataCluster/dialog/bdcDashboardModel.ts]: ClusterController throws an error in the constructor if the username or password is empty and basic auth is selected

File: extensions/big-data-cluster/src/bigDataCluster/dialog/bdcDashboardModel.ts (hunk 3 above)
Comment (Charles-Gagnon) [reply extensions/big-data-cluster/src/bigDataCluster/dialog/bdcDashboardModel.ts]: The dashboard is notified that an error has occurred but right now we don't handle that in any special way so nothing happens (it sits there spinning). The user can press refresh to cause the dialog to appear again and if the correct information is entered then the dashboard will refresh with new information as expected.

I have a follow up task to make the dashboard indicate if an error occurred but that won't be a part of this PR.

**Comment by Charles-Gagnon** [comment]
@kevcunnane I changed it as discussed earlier so it now keeps the password for the session (re-prompting if the credentials are wrong though). 

I also fixed up some no-floating-promises errors

**Pull request merged**